src/storage/clickhouse/clickhouse.constants.ts

```diff
@@ -158,7 +159,40 @@ export const validatorCountWithMissAttestationLastNEpochQuery = (
     )
     WHERE count_fail = ${epochInterval}
     GROUP BY nos_name
-    ORDER BY miss_attestation_count DESC
+    ORDER BY suitable DESC
+  `;
+};
+
+export const validatorCountHighAvgIncDelayAttestationOfNEpochQuery = (
+  fetchInterval: number,
+  slot: string,
+  epochInterval: number,
+): string => {
+  return `
+    SELECT
+      nos_name,
+      count() as suitable
+    FROM (
+      SELECT
+        validator_pubkey,
+        avg(inclusion_delay) as avg_inclusion_delay,
+        nos_name
+      FROM (
+        SELECT
+          validator_pubkey,
+          nos_name,
+          inclusion_delay
+        FROM stats.validator_attestations
+        WHERE
+          (slot_to_attestation <= ${slot} AND slot_to_attestation > (${slot} - ${fetchInterval} * ${epochInterval}))
+        ORDER BY slot_to_attestation DESC, validator_pubkey
+      )
+      GROUP BY validator_pubkey, nos_name
+      ORDER BY avg_inclusion_delay DESC, validator_pubkey
+    )
+    WHERE avg_inclusion_delay > 2
```

Comment:
Use `HAVING` to remove one level.

Reply:
Didn't find a place for it

Reply:
Something like this (NOT TESTED)?

```sql
SELECT
  nos_name,
  count() as suitable
FROM (
  SELECT
    validator_pubkey,
    avg(inclusion_delay) as avg_inclusion_delay,
    nos_name
  FROM stats.validator_attestations
  WHERE
    (slot_to_attestation <= ${slot} AND slot_to_attestation > (${slot} - ${fetchInterval} * ${epochInterval}))
  GROUP BY validator_pubkey, nos_name
  HAVING avg_inclusion_delay > 2
  ORDER BY avg_inclusion_delay DESC, slot_to_attestation DESC, validator_pubkey
)
GROUP BY nos_name
ORDER BY suitable DESC
```

Reply:
I tested several variants (including yours) and no one of them brought expected result

Reply:
Fixed a mistake in the proposed query and it returns the same data on my test:

```sql
SELECT
  nos_name,
  count() as suitable
FROM (
  SELECT
    validator_pubkey,
    avg(inclusion_delay) as avg_inclusion_delay,
    nos_name
  FROM stats.validator_attestations
  WHERE
    (slot_to_attestation <= ${slot} AND slot_to_attestation > (${slot} - ${fetchInterval} * ${epochInterval}))
  GROUP BY validator_pubkey, nos_name
  HAVING avg_inclusion_delay > 2
  ORDER BY avg_inclusion_delay DESC, validator_pubkey
)
GROUP BY nos_name
ORDER BY suitable DESC;
```

Reply:
done